components/molecules/NivoScatterChart/nivo-scatter-chart.tsx

```diff
@@ -27,21 +35,42 @@ interface ScatterPlotProps {
   }[];
   isMobile?: boolean;
   repositories?: number[];
+  metadata: ScatterChartMetadata;
+  handleSetPrFilter: (state: "open" | "closed" | "all") => void;
 }
+
 const NivoScatterPlot = ({
   data,
   maxFilesModified,
   title,
   setShowBots,
   showBots,
   isMobile,
-  repositories
+  repositories,
+  metadata,
+  handleSetPrFilter
 }: ScatterPlotProps) => {
   const [showMembers, setShowMembers] = useState<boolean>(false);
   const [isLogarithmic, setIsLogarithmic] = useState<boolean>(false);
 
   let functionTimeout: any;
 
+  const handleTogglePrFilter = (val: string) => {
+    switch (val) {
```

Comment:
IMHO `switch` statements it is very difficult, if you want to avoid `switch` it would be better

Reply:
In this case it is atleast in my view much more developer friendly to be able to quickly & easily see the possible states the value could take, instead of a multiple if-elses.